src/Microsoft.Health.Fhir.Liquid.Converter/Cda/CdaDataParser.cs

```diff
@@ -0,0 +1,122 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
+using Microsoft.Health.Fhir.Liquid.Converter.Exceptions;
+using Microsoft.Health.Fhir.Liquid.Converter.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter.Cda
+{
+    public class CdaDataParser
+    {
+        public IDictionary<string, object> Parse(string document)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(document))
+                {
+                    throw new DataParseException(FhirConverterErrorCode.NullOrEmptyInput, Resources.NullOrEmptyInput);
+                }
+
+                // Remove line breaks to avoid invalid line breaks in json value
+                document = Regex.Replace(document, @"\r\n?|\n", string.Empty);
+
+                var xDocument = XDocument.Parse(document);
+
+                // Remove redundant namespaces to avoid appending namespace prefix before elements
+                var defaultNamespace = xDocument.Root?.GetDefaultNamespace().NamespaceName;
+                xDocument.Root?.Attributes()
+                    .Where(attribute => IsRedundantNamespaceAttribute(attribute, defaultNamespace))
+                    .ToList()
+                    .ForEach(action: x => x.Remove());
+
+                // Normalize non-default namespace prefix in elements
+                var namespaces = xDocument.Root?.Attributes()
+                    .Where(x => x.IsNamespaceDeclaration && x.Value != defaultNamespace);
+                NormalizeNamespacePrefix(xDocument?.Root, namespaces);
+
+                // Change XText to XElement with name "_" to avoid serializing depth difference, e.g., given="foo" and given.#text="foo"
+                ReplaceTextWithElement(xDocument?.Root);
+
+                // Convert to json dictionary
+                var jsonString = JsonConvert.SerializeXNode(xDocument);
+                var dataDictionary = JsonConvert.DeserializeObject<IDictionary<string, object>>(jsonString, new DictionaryJsonConverter()) ??
+                                     new Dictionary<string, object>();
```

Comment:
Maybe we can deserialize Xml document to dictionary directly (somewhat improve efficiency?)